src/overlays/actors/ovl_Obj_Mine/z_obj_mine.c

```diff
@@ -24,131 +56,1178 @@ ActorInit Obj_Mine_InitVars = {
+
+    if (type == OBJMINE_TYPE_PATH) {
+        ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 45.0f);
+        this->actor.shape.shadowAlpha = 140;
+        this->pathSpeed = sPathSpeeds[OBJMINE_GET_PATH_SPEED(&this->actor)]; // sPathSpeeds[i] = i + 1.0f
+        if (pathIndex == 0xFF) {
+            ObjMine_Path_SetupStationary(this);
+        } else {
+            path = &play->setupPathList[pathIndex];
+
+            this->waypointIndex = 0;
+            this->waypointCount = path->count - 1;
+            this->waypoints = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+
+            ObjMine_Path_MoveToWaypoint(this, this->waypointIndex);
+            ObjMine_Path_SetupMove(this);
+        }
+        Matrix_SetTranslateRotateYXZ(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
+                                     &this->actor.shape.rot);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+        Collider_UpdateSpheres(0, &this->collider);
+        this->actor.floorHeight = BgCheck_EntityRaycastFloor5(&play->colCtx, &this->actor.floorPoly, &dummyBgId,
+                                                              &this->actor, &this->actor.world.pos);
+    } else {
+        s32 linkCount = OBJMINE_GET_LINK_COUNT(&this->actor);
+
+        this->actor.update = ObjMine_AirWater_Update;
+        this->actor.uncullZoneScale = 150.0f + (linkCount * (LINK_SIZE * 1.75f));
+        this->actor.uncullZoneDownward = 150.0f + (linkCount * (LINK_SIZE * 1.75f));
+        ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 45.0f);
+        this->actor.shape.shadowAlpha = 140;
+
+        if (type == OBJMINE_TYPE_AIR) {
+            this->actor.draw = ObjMine_Air_Draw;
+            ObjMine_Air_InitChain(this, linkCount);
+            this->actor.world.pos.y = -ATTACH_OFFSET - (linkCount * LINK_SIZE) + this->actor.home.pos.y;
+            ObjMine_Air_InitCollider(this, linkCount);
+            func_800B4AEC(play, &this->actor, 0.0f);
+            if (linkCount == 0) {
+                ObjMine_Air_SetupStationary(this);
+            } else {
+                ObjMine_Air_SetupChained(this);
+            }
+        } else {
+            this->actor.draw = ObjMine_Water_Draw;
+            ObjMine_Water_InitChain(this, linkCount);
+            this->actor.world.pos.y = ATTACH_OFFSET + (linkCount * LINK_SIZE) + this->actor.home.pos.y;
+            ObjMine_Water_InitCollider(this, linkCount);
+            if (linkCount == 0) {
+                ObjMine_Water_SetupStationary(this);
+            } else {
+                ObjMine_Water_SetupChained(this);
+            }
+        }
+    }
+}
+
+void ObjMine_Destroy(Actor* thisx, PlayState* play) {
+    ObjMine* this = THIS;
+    // PlayState* play = play2; // must be in this order to match with recast
+
+    Collider_DestroyJntSph(play, &this->collider);
```

Comment:
```suggestion
void ObjMine_Destroy(Actor* thisx, PlayState* play) {
    ObjMine* this = THIS;

    Collider_DestroyJntSph(play, &this->collider);
```

Reply:
If we ever get around to standardizing the whole gamestate/playstate thing, might be worth noting

Reply:
Fair point. We can deal with it later.

Reply:
Yeah, it's not particularly useful to have scattered isolated comments inconsistently when it's something we have to look at repo-wide, we can make a note of it somewhere but putting it in the repo makes it feel like a bit more definitive.